aws/src/index.js

```diff
@@ -349,9 +350,37 @@ class S3HttpCollector extends Collector {
 }
 
 /* eslint-disable no-param-reassign */
-const processLogText = (cloudWatchLogs, tagRegexMap) => {
+const processLogText = (cloudWatchLogs, tagRegexMap, arn) => {
+  // derive source
+  const keys = arn.split(':');
+  const region = keys[3];
+  const accountId = keys[4];
+
+  // default value
+  let source = accountId + '-' + region;
+
+  // derive source for lambda function or EC2 instance
+  const logGroup = cloudWatchLogs.logGroup;
+  const logStream = cloudWatchLogs.logStream;
+
+  if(logGroup.startsWith('/aws/lambda/')) {
+    const logGroupKeys = logGroup.split('/');
+    source = logGroupKeys[logGroupKeys.length - 1];
+  } else if (logStream.startsWith('i-')) {
+    const logStreamKeys = logGroup.split('-');
+    source = logStreamKeys[logStreamKeys.length - 1];
+  }
+
   for (const logEvent of cloudWatchLogs.logEvents) {
     logEvent.log_type = 'aws_cloud_watch';
+    logEvent.source = source;
+    logEvent.accoundId = accountId;
+    logEvent.Region = region;
```

Comment:
logEvent.Region -> logEvent.region, can we use 'r'?

Reply:
To correlate logs with metrics we will need to add a tag `Region`(with `R`). Cloud metrics add this tag as well.